src/org/rascalmpl/util/maven/SimpleRepositoryDownloader.java

```diff
@@ -50,46 +60,142 @@ public SimpleRepositoryDownloader(Repository repo, HttpClient client) {
     }
 
     public boolean download(String url, Path target, boolean force) {
-        // TODO also download md5/sha256 files to verify we got the right files.
-        // see here on the information in headers instead of a separate file request
-        // https://maven.apache.org/resolver/expected-checksums.html
-        // calculate it during file writing (we have to override BodyHandlers)
+        Optional<Path> result = download(url, target, force,
+        (InputStream input) -> { 
+            Path tempArtifact = Files.createTempFile("maven-download", null);
+            Files.copy(input, tempArtifact, StandardCopyOption.REPLACE_EXISTING);
+            return tempArtifact;
+        },
+        (Path tempArtifact) -> {
+            boolean copyResult = copyFileToTarget(target, force, tempArtifact);
+            cleanup(tempArtifact);
+            return copyResult;
+        });
+        return result.isPresent();
+    }
+
+    public String downloadPom(String url, Path target, boolean force) {
```

Comment:
nit: can be called `downloadAndRead` or something? I mean, it doesn't have to be only used for pom files.